### ISSUE TITLE
(question) eslint rule / plugin for guideline 7.15

### ISSUE THREAD

**Issue by shelbyspeegle** [airbnb/javascript#1731]
is there an eslint rule or plugin that enforces [7.15](https://github.com/airbnb/javascript#functions--signature-invocation-indentation)?

**Comment by ljharb** [comment]
Indeed! `comma-dangle` handles the trailing comma, and `function-paren-newline`, i believe, handles the rest.

`comma-dangle` should be linked here already; however, I'm still not 100% sure about `function-paren-newline` :-)

**Comment by shelbyspeegle** [comment]
awesome! thanks @ljharb.

i can't seem to find the right options for `function-paren-newline` that adheres to the airbnb eslint guideline 7.15.

in particular, i can't get eslint to bug me about the 'bad' example below.

```
// bad
console.log(foo,
  bar,
  baz);

// good
console.log(
  foo,
  bar,
  baz,
);
```

**Comment by ljharb** [comment]
Maybe we need to file something upstream on eslint about it?

**Comment by raunofreiberg** [comment]
@shelbyspeegle I tried both rules out and I'm getting both errors, the bad example included.

```
"comma-dangle": "error",
"function-paren-newline": "error"
```

What have you tried so far?

**Comment by shelbyspeegle** [comment]
thanks @raunofreiberg, that gets me close, unfortunately it allows this scenario:

```
console.log(
    foo,
    bar, baz,
    yo,
  );
```

i assumed that this _should_ produce an eslint error, but an example isnt provided for this type of case.

i did see this in the guide language that could suggest that this should error.
> with each item on a line by itself

**Comment by ljharb** [comment]
Yes, that should be an error. Either the entire invocation fits on one line, or each argument is on a line by itself.

**Comment by GusBuonv** [comment]
Doesn't the "multiline" option on [function-paren-newline](https://eslint.org/docs/rules/function-paren-newline) along with the current "comma-dangle" implement [7.15](https://github.com/airbnb/javascript#functions--signature-invocation-indentation) correctly? If so [this line](https://github.com/airbnb/javascript/blob/f5c14cae2ff58000cead98290b8ec4b54dda2f14/packages/eslint-config-airbnb-base/rules/style.js#L106) should be updated.

**Comment by shelbyspeegle** [comment]
@GusBuonv That works great for me!

For this test file, eslint errors show:
![image](https://user-images.githubusercontent.com/7694692/133004114-edce13a3-cc1d-480d-85db-163c3df451b2.png)


... and `eslint --fix` now properly resolves that to:
![image](https://user-images.githubusercontent.com/7694692/133004123-2beb9f80-ed07-41b4-b13e-e0d35ac3ae28.png)


Using your suggested rules in this simple `eslintrc.js`.
```
module.exports = {
  extends: ['airbnb'],
  rules: {
    'function-paren-newline': ['error', 'multiline'],
  },
};
```

Thanks @GusBuonv!

**Comment by ljharb** [comment]
Reading the rule docs, it does seem like `multiline-arguments` is the option that would most closely match the guide's intent.

It was changed from "multiline" to "consistent" [in 2018](https://github.com/airbnb/javascript/commit/b56875bee21eb8d87213c787dc105c288066f43e). eslint itself added `multiline-arguments` [in 2019](https://github.com/eslint/eslint/pull/11406), so it seems like "consistent" was better than "multiline", but "multiline-arguments" may be better still.

I'm hesitant to make this change, however, until it's at least been verified on airbnb's own code. I'll reach out to folks there and see how it goes.

**Comment by ljharb** [comment]
Given that this rule is disabled by eslint-config-prettier, Airbnb won't be affected, so I think this is a reasonable semver-minor change to make. @GusBuonv please make that PR :-)

**Comment by GusBuonv** [comment]
@ljharb Done! If the PR is merged, it may still be a good idea to keep this issue open or open a new one. I assume [7.15](https://github.com/airbnb/javascript#functions--signature-invocation-indentation) is also supposed to apply to argument destructuring. If it is meant to apply in this case, it is still not properly enforced, and current core ESLint rules don't provide a way to achieve the desired behavior.
